content/post/2023-06-06-system-dependencies/index.Rmd

```diff
@@ -0,0 +1,295 @@
+
+``` diff
+jobs:
+  R-CMD-check:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
++      - name: Install extra system dependencies manually
++        run:
++          wget ...
++          make
++          sudo make install
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
+
+      - uses: r-lib/actions/check-r-package@v2
+```
+
+You can see [a real-life example in the rbi R package](https://github.com/sbfnk/rbi/blob/9b05a24ce42f7b1b53481370f3bde3dcd86bca02/.github/workflows/R-CMD-check.yaml).
+
+#### Using a Docker image in GitHub Actions
+
+Alternatively, you can do the manual installation in a Docker image and use this image in your GitHub Actions workflow.
+This is a particularly good solution if there is already a public Docker image or you already wrote a `DOCKERFILE` for your own local development purposes.
+If you use a public image, you can follow [the steps in the official documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-running-a-job-within-a-container) to integrate it to your GitHub Actions job.
+If you use a `DOCKERFILE`, you can follow [the answers to this stackoverflow question](https://stackoverflow.com/q/61154750/4439357) (in a nutshell, use `docker compose` in your job or publish the image first and then follow the official documentation).
+
+``` diff
+jobs:
+  R-CMD-check:
+    runs-on: ubuntu-latest
++    container: ghcr.io/org/repo:main
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
+
+      - uses: r-lib/actions/check-r-package@v2
+```
+
+You can again see [a real-life example in the rbi R package](https://github.com/sbfnk/rbi/pull/46/files).
+
+## Conclusion
+
+In this post, we have provided an overview of how to specify system requirements for R package, how this seemingly innocent task requires a very complex infrastructure so that it can be understood by automated tools and that your dependencies are smoothly installed in a single command.
+We also gave some pointers on what to do if you're in one of the rare cases where the automated tools don't or can't work.
+
+One final note on this topic is that there might be a move from CRAN to start requiring more standardization in the `SystemRequirements` field.
+One R package developer has reported being asked to change "Java (\>= 8)" to "Java JRE 8 or higher".
```

Comment:
```suggestion
One R package developer has reported being asked to change "Java JRE 8 or higher" to "Java (\>= 8)".
```
If I read the message correctly, it's the other way round.

Reply:
You're completely right, thanks! I've read this message so many times. I'm not sure how I manage to understand the opposite :facepalm: